[Docs] Feature maps and QML reorganization (#92)
pasqal-io/qadence: docs/qml/ml_tools.md

to build dictionaries of `DataLoader`s instances and easily iterate over them.

```python exec="on" source="material-block"
import torch
from torch.utils.data import DataLoader, TensorDataset
from qadence.ml_tools import DictDataLoader

def dataloader() -> DataLoader:
    batch_size = 5
    x = torch.linspace(0, 1, batch_size).reshape(-1, 1)
    y = torch.sin(x)

    dataset = TensorDataset(x, y)
    return DataLoader(dataset, batch_size=batch_size)

def dictdataloader() -> DictDataLoader:
    batch_size = 5

    keys = ["y1", "y2"]
    dls = {}
    for k in keys:
        x = torch.rand(batch_size, 1)
        y = torch.sin(x)
        dataset = TensorDataset(x, y)
        dataloader = DataLoader(dataset, batch_size=batch_size)
        dls[k] = dataloader

    return DictDataLoader(dls)

n_epochs = 2

# iterate standard DataLoader
dl = dataloader()
for i in range(n_epochs):
    data = next(iter(dl))

# iterate DictDataLoader
ddl = dictdataloader()
for i in range(n_epochs):
    data = next(iter(ddl))

```

## Optimization routines

For training QML models, Qadence also offers a few out-of-the-box routines for optimizing differentiable

> models like `QNN`s and `QuantumModel`s containing either *trainable* and/or *non-trainable* parameters

Reviewer: ```suggestion
models, _e.g._ `QNN` and `QuantumModel`, containing either *trainable* and/or *non-trainable* parameters
```

> (you can refer to [the parameters tutorial](../tutorials/parameters.md) for a refresh about different parameter types):

Reviewer: ```suggestion
(see [the parameters tutorial](../tutorials/parameters.md) for detailed information about parameter types):
```
